cmd/awscollector/main.go

```diff
@@ -46,8 +47,22 @@ func main() {
 		log.Fatalf("failed to build components: %v", err)
 	}
 
+	var zapHooks []zap.Option
+
 	// init lumberFunc for zap logger
 	lumberHook := logger.GetLumberHook()
+	if lumberHook != nil {
+		zapHooks = append(zapHooks, zap.Hooks(lumberHook))
+	}
+
+	// init an ECS Error Logger and a go routine for error reporting when the STATUS_MESSAGE_FILE_PATH is set
+	if os.Getenv("STATUS_MESSAGE_FILE_PATH") != "" {
+		ecsErrorLogger := logger.NewECSErrorLogger()
```

Comment:
If the main routine is going to be responsible for determining whether this logger should be used based on environment variables then it should provide the entire configuration to the logger.  In other words, exactly one of the main routine and the logger should be aware of the existence of these environment variables.  I'm not sure that putting it in the main routine is the correct choice, but I am sure that splitting it between the two is the wrong choice.